client/tests/simulator_tests.rs

```diff
@@ -83,6 +85,66 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
+fn test_get_random_max_size() {
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    let command = GetRandomCmd {
+        bytes_requested: 4096,
+    };
+
+    // We allow panic in test cases.
+    let resp = run_command(&command, &mut tpm).expect("Failed running command.");
+
+    // Lets pull out the actual data as a slice for convenience
+    let random_slice = &resp.random_bytes.buffer[..resp.random_bytes.size as usize];
+
+    // We should have received less than what the client implementation supports.
+    assert!(random_slice.len() <= TpmtHa::UNION_SIZE);
```

Comment:
Optional: Instead of having the comments here, you could mix them into the assert so that they print on failure.

e.g. `assert!(..., "We expected to have received no more than the maximum supported digest size")`

Reply:
Fixed.